user-service/src/main/java/com/movie/recommender/user/config/SecurityConfig.java

```diff
@@ -21,7 +21,10 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                 .sessionManagement(session -> session
                         .sessionCreationPolicy(SessionCreationPolicy.STATELESS))
                 .authorizeHttpRequests(authorize -> authorize
-                        .requestMatchers("/api/v1/users/register", "/api/v1/users/login").permitAll()
+                        .requestMatchers("/api/v1/users/register",
+                                "/api/v1/users/login",
+                                "/api/v1/favorites/{userId}",
+                                "/api/v1/favorites/{userId}/{movieId}").permitAll()
```

Comment:
path variables like `{userId}` and `{movieId}` cannot be used in `requestMatchers`
instead, use 
`.requestMatchers("/api/v1/favorites/**").permitAll()` 
this matches the base path and any additional parameters without explicitly listing each level